aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/CreateHandler.java

```diff
@@ -0,0 +1,46 @@
+package software.amazon.logs.querydefinitions;
+
+import software.amazon.awssdk.services.cloudwatchlogs.model.InvalidParameterException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutQueryDefinitionResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ServiceUnavailableException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class CreateHandler extends BaseHandler<CallbackContext> {
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+        PutQueryDefinitionResponse putQueryDefinitionResponse;
```

Comment:
I sometimes like putting these try/catch throw blocks into private methods so that you avoid the unassigned declaration which IMO sometimes makes it unclear where the value is being set. e.g.
```
PutQueryDefinitionResponse putQueryDefinitionResponse = invoke(/*...*/);
```

Reply:
ah good call out. I'll try that out